Xamarin.Forms.Platform.iOS/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,475 @@
+			AddSubview(_avPlayerViewController.View);
+		}
+
+		VisualElement IVisualElementRenderer.Element => MediaElement;
+
+		UIView IVisualElementRenderer.NativeView => this;
+
+		UIViewController IVisualElementRenderer.ViewController => _avPlayerViewController;
+		
+		bool _idleTimerDisabled = false;
+		void SetKeepScreenOn(bool value)
+		{
+			if (value)
+			{
+				if (!UIApplication.SharedApplication.IdleTimerDisabled)
+				{
+					_idleTimerDisabled = true;
+					UIApplication.SharedApplication.IdleTimerDisabled = true;
+				}
+			}
+			else
+			{
+				if (_idleTimerDisabled)
+				{
+					_idleTimerDisabled = false;
+					UIApplication.SharedApplication.IdleTimerDisabled = false;
+				}
+			}
+		}
+
+		void UpdateSource()
+		{
+			if (MediaElement.Source != null)
+			{
+				AVAsset asset = null;
+				
+				var uriSource = MediaElement.Source as UriMediaSource;
+				if (uriSource != null)
+				{
+					if (uriSource.Uri.Scheme == "ms-appx")
+					{
+						// used for a file embedded in the application package
+						asset = AVAsset.FromUrl(NSUrl.FromFilename(uriSource.Uri.LocalPath.Substring(1)));
+					}
+					else if (uriSource.Uri.Scheme == "ms-appdata")
+					{
+						string filePath = string.Empty;
+
+						if (uriSource.Uri.LocalPath.StartsWith("/local"))
+						{
+							filePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), uriSource.Uri.LocalPath.Substring(7));
+						}
+						else if (uriSource.Uri.LocalPath.StartsWith("/temp"))
+						{
+							filePath = Path.Combine(Path.GetTempPath(), uriSource.Uri.LocalPath.Substring(6));
+						}
+						else
+						{
+							throw new ArgumentException("Invalid Uri", "Source");
+						}
+
+						asset = AVAsset.FromUrl(NSUrl.FromFilename(filePath));
+					}
+					else
+					{
+						asset = AVUrlAsset.Create(NSUrl.FromString(uriSource.Uri.AbsoluteUri));
+					}
+				}
+				else
+				{
+					var fileSource = MediaElement.Source as FileMediaSource;
+					if (fileSource != null)
+					{
+						asset = AVAsset.FromUrl(NSUrl.FromFilename(fileSource.File));
+					}
+				}
+
+				
+				AVPlayerItem item = new AVPlayerItem(asset);
+				RemoveStatusObserver();
+
+				_statusObserver = (NSObject)item.AddObserver("status", NSKeyValueObservingOptions.New, ObserveStatus);
+				
+				
+				if (_avPlayerViewController.Player != null)
+				{
+					_avPlayerViewController.Player.ReplaceCurrentItemWithPlayerItem(item);
+				}
+				else
+				{
+					_avPlayerViewController.Player = new AVPlayer(item);
+					_rateObserver = (NSObject)_avPlayerViewController.Player.AddObserver("rate", NSKeyValueObservingOptions.New, ObserveRate);
+				}
+				
+				if (MediaElement.AutoPlay)
+				{
+					var audioSession = AVAudioSession.SharedInstance();
+					NSError err = audioSession.SetCategory(AVAudioSession.CategoryPlayback);
+					audioSession.SetMode(AVAudioSession.ModeMoviePlayback, out err);
+					err = audioSession.SetActive(true);
```

Comment:
Possible errors are ignored. Perhaps the best solution would be to add an error event to the `MediaElement` class?